.github/workflows/unit-tests.yml

```diff
@@ -223,3 +223,59 @@ jobs:
             echo "All test jobs have completed successfully or been skipped!"
             exit 0
           fi
+
+  push-to-gitlab-registry:
```

Comment:
Couldn't we just add a tag to `build-push-action`? Maybe add a separate docker-metadata-action that only runs on push events if we want to get out the sha. I'm not sure we need to separate this out as a separate step